engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/JoinGameScreen.java

```diff
@@ -416,7 +416,12 @@ public String get() {
     private String getWorldText(Future<ServerInfoMessage> info) {
         try {
             List<String> codedWorldInfo = new ArrayList<>();
-            for (WorldInfo wi : info.get().getWorldInfoList()) {
+            ServerInfoMessage serverInfoMessage = info.get();
+
+            if(serverInfoMessage == null)
+                return FontColor.getColored(translationSystem.translate("${engine:menu#connection-failed}"), Color.RED);
```

Comment:
Please put brackets around the body of if.

Here at Terasology, we use brackets around if/while even if it's a single line.

Reply:
should be fine now
I also didnt write catch clause for this, because it could catch some more general exceptions. Hope it is right like that